### PR TITLE
Added Eigen dependency (GRIDEDIT-602)

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: cmake/fetch_content.cmake [hunk 1]
```diff
@@ -83,4 +83,20 @@ if(${USE_LIBFMT})
...
-  
\ No newline at end of file
+  
+# Eigen
+# Note: v3.4.0 seems to have a problem detecting c++11 when MSVC is used, so master will be used here.
+FetchContent_Declare(
+  Eigen
+  GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
+  GIT_TAG        master
+)
```
Comment (BillSenior) [inline cmake/fetch_content.cmake:93]: Could we have unit test failures when somebody makes a commit to the Eigen library if we always the head of the master?
Would it be an idea to use a specific commit to the master?

Comment (BillSenior) [reply cmake/fetch_content.cmake:93]: On a previous project somebody made a change to the testing environment that it always used the latest version of the Intel c++ compiler (why not, seems like a good idea)(we did not do much in the way of code reviewing at the time). Then one day, we had failures in many of the tests. We spent a a long time looking for the problem, undoing work getting really confused why we had these failures. We eventually found the cause, a system admin somewhere (not in our company) made an update to the Intel compiler.

File: libs/MeshKernel/include/MeshKernel/Entities.hpp [hunk 2]
```diff
@@ -159,7 +161,17 @@ namespace meshkernel
...
+
+    /// @brief Row major dynamic matrix
+    /// @tparam T Data type
+    template <class T>
+    using MatrixRowMajor = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+
+    /// @brief Column major dynamic matrix
+    /// @tparam T Data type
+    template <class T>
+    using MatrixColMajor = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::ColMajor>;
```
[the remark this answers is not in the document]
Comment (BillSenior) [reply libs/MeshKernel/include/MeshKernel/Entities.hpp:174]: Might consider a nested namespace, e.g. linalg?

File: cmake/fetch_content.cmake (hunk 1 above)
Comment (ahmad-el-sayed) [reply cmake/fetch_content.cmake:93]: That's a very good point. I'll check out a commit in a working state instead.

File: libs/MeshKernel/include/MeshKernel/Entities.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply libs/MeshKernel/include/MeshKernel/Entities.hpp:174]: Sure can. The project in general lacks namespaces.

File: cmake/fetch_content.cmake (hunk 1 above)
Comment (ahmad-el-sayed) [reply cmake/fetch_content.cmake]: Done

File: libs/MeshKernel/include/MeshKernel/Entities.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply libs/MeshKernel/include/MeshKernel/Entities.hpp]: This has been moved to the dedicated header `libs/MeshKernel/include/MeshKernel/Utilities/LinearAlgebra.hpp`

**Review by BillSenior — APPROVED** [review]
Looks good

**Pull request merged**